app/CLI/Parsers.hs

```diff
@@ -34,35 +35,55 @@ import Options.Applicative (
  )
 
 -- Cardano node imports
-import Cardano.Api (NetworkMagic (..), fromNetworkMagic)
+import Cardano.Api (NetworkId, NetworkMagic (..), fromNetworkMagic)
 
 -- Hydra auction imports
 import HydraAuction.Fixture (Actor (..))
 import HydraAuction.OnChain (AuctionScript (..))
 import HydraAuction.Types (Natural, intToNatural)
 
 -- Hydra auction CLI imports
-import CLI.Actions (CliAction (..), CliInput (..), seedAmount)
+import CLI.Actions (CliAction (..), seedAmount)
 import CLI.Config (AuctionName (..))
+import Cardano.Prelude (asum)
 
-parseCliAction :: [String] -> Either String CliAction
-parseCliAction s = case execParserPure preferences options s of
-  Success a -> Right a
-  Failure failure -> Left $ fst $ renderFailure failure ""
-  _ -> Left "error"
+data CliInput
+  = Watch !AuctionName
+  | InteractivePrompt !PromptOptions
+
+data PromptOptions = MkPromptOptions
+  { cliActor :: !Actor
+  , cliVerbosity :: !Bool
+  , cliNodeSocket :: !String
+  , cliNetworkId :: !NetworkId
+  }
+
+getCliInput :: IO CliInput
+getCliInput = customExecParser preferences options
   where
     options =
       info
-        (cliActionParser <**> helper)
+        (cliInputParser <**> helper)
         fullDesc
     preferences = prefs (showHelpOnEmpty <> showHelpOnError)
 
-getCliInput :: IO CliInput
-getCliInput = customExecParser preferences options
+cliInputParser :: Parser CliInput
+cliInputParser =
+  asum
+    [ Watch <$> watchAuction
+    , InteractivePrompt
+        <$> (MkPromptOptions <$> actor <*> verboseParser <*> socketDir <*> (fromNetworkMagic <$> networkMagic))
+    ]
+
+parseCliAction :: [String] -> Either String CliAction
+parseCliAction s = case execParserPure preferences options s of
+  Success a -> Right a
+  Failure failure -> Left $ fst $ renderFailure failure ""
```

Comment:
Why is the help an empty string?

Reply:
Oh, I think the parser prefs already setup the error reporting.